File: common/src/test/java/se/casaferre/common/VolumeTest.java
package se.casaferre.common;


import org.junit.Test;
import static org.assertj.core.api.Assertions.assertThat;

public class VolumeTest {

    @Test
    public void testFromLiter() {
        Volume volume = Volume.setLiter(100.0);
        assertThat(volume.getUsGallon()).isBetween(26.41721, 26.41722);
        assertThat(volume.getUkGallon()).isBetween(21.99692, 21.99693);
    }

    @Test
    public void testFromUsGallon() {
        Volume volume = Volume.setUsGallon(100.0);
        assertThat(volume.getLiter()).isBetween(378.54, 378.55);
        assertThat(volume.getUkGallon()).isBetween(83.26737, 83.26738);
    }

    @Test
    public void testFromUkGallon() {
        Volume volume = Volume.setUkGallon(100.0);
        assertThat(volume.getLiter()).isBetween(454.60, 454.61);
        assertThat(volume.getUsGallon()).isBetween(120.09504, 120.09505);
    }
}

File: common/src/test/kotlin/se/casaferre/common/VolumeTest.kt
package se.casaferre.common

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class VolumeTest {

    @Test
    fun generalTest() {
        Assertions.assertEquals(Volume.liter2usGallon(100.0),26.41721, 0.0001)
        Assertions.assertEquals(Volume.liter2imperialGallon(100.0),21.99692, 0.0001)
        Assertions.assertEquals(Volume.usGallon2liter(100.0),378.54, 0.0001)
        Assertions.assertEquals(Volume.imperialGallon2liter(100.0),454.60, 0.0001)
    }

}